snuba/snuba_migrations/replays/0019_add_materialization.py

```diff
@@ -0,0 +1,182 @@
+from typing import Iterator, List, Sequence
+
+from snuba.clickhouse.columns import (
+    UUID,
+    AggregateFunction,
+    Column,
+    DateTime,
+    IPv4,
+    IPv6,
+    String,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return list(forward_iter())
+
+    def backwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return list(backward_iter())
+
+
+def forward_iter() -> Iterator[operations.SqlOperation]:
+    yield operations.CreateTable(
+        storage_set=StorageSetKey.REPLAYS,
+        table_name="replays_aggregated_local",
+        columns=columns,
+        engine=table_engines.AggregatingMergeTree(
+            storage_set=StorageSetKey.REPLAYS,
+            order_by="(project_id, toStartOfHour(to_hour_timestamp), replay_id)",
```

Comment:
I think you might need `retention_days` in the `order_by`, but then you could add a `primary_key` attribute with just these three columns.

Reply:
I think it needs to be in the primary key right?  In case the same replay has multiple retention periods we'd want the materialized rows to expire at the same time as the source table.

At the moment we only support one retention period but I figure its better to preserve this than not.

Reply:
That works too.